### ISSUE TITLE
any support of k-fold cross validation?

### ISSUE THREAD

**Issue by Emadeldeen-24** [victoresque/pytorch-template#65]

**Comment by LumosWC** [comment]
yeah please. Or any suggestions on how to modify the code if I want to use k-fold cross validation please?

**Comment by SunQpark** [comment]
Sorry for late response. K-fold CV can be done by modifying `BaseDataLoader`.

Current implementation of `_split_sampler` use first n samples of dataset as validation set. Modify this function to take additional argument, `fold_idx` and to use specified part of dataset.
To be more specific, [this part](https://github.com/victoresque/pytorch-template/blob/a39a102914feace71ddd05b58d5e5a7c0e2c99dd/base/base_data_loader.py#L45-L46) should be changed to following.
```python
valid_idx = idx_full[len_valid*fold_idx, len_valid*(fold_idx+1)]
train_idx = np.delete(idx_full, np.arange(0, len_valid))
```
Then, add the `fold_idx` to `BaseDataLoader.__init__` and `YourDataLoader.__init__`, specify it in the config file, under `data_loader.args`.

**Comment by james60708** [comment]
@SunQpark I think you mean
```
valid_idx = idx_full[len_valid * fold_idx : len_valid * (fold_idx + 1)]
train_idx = np.delete(idx_full, np.arange(len_valid * fold_idx, len_valid * (fold_idx + 1)))
```

**Comment by deeperlearner** [comment]
I have implemented cross validation to my [Pytorch-Template](https://github.com/deeperlearner/Pytorch-Template)
See #88
